### PR TITLE
[CDF-25640] 😄Chart StorageIO

### PR DESCRIPTION
**Pull request by doctrino**
# Description

This PR introduces Chart implementation of the StoreageIO interface. This is a prerequisite for adding Chart upload/download command, and it will likely be used in a `cdf migrate charts` command.

## Changelog

- [ ] Patch
- [x] Skip

### CODE REVIEW

**Review by Aashutosh-cognite — APPROVED** [review]

**Review by MyBitBird — COMMENTED** [review]
one comment and one question, need to go through it again after getting help with the docs

File: cognite_toolkit/_cdf_tk/storageio/_applications.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    supported_compressions = frozenset({".gz"})
+    supported_read_formats = frozenset({".ndjson"})
+    chunk_size = 10
+
+    def download_iterable(self, selector: ChartSelector, limit: int | None = None) -> Iterable[ChartList]:
+        selected_charts = self.client.charts.list(visibility="PUBLIC")
+        if isinstance(selector, AllChartSelector):
+            ...
+        elif isinstance(selector, ChartOwnerSelector):
+            selected_charts = ChartList([chart for chart in selected_charts if chart.owner_id == selector.owner_id])
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite_toolkit/_cdf_tk/storageio/_applications.py:29]: It is a hidden API: https://api-docs.cogheim.net/operations/listEntities

Only used by the Charts frontend in fusion.

File: cognite_toolkit/_cdf_tk/client/_toolkit_client.py [hunk 2]
```diff
@@ -92,6 +93,26 @@ def create_api_url(self, endpoint: str) -> str:
...
+        Returns:
+            str: The full App URL.
+
+        Examples:
+            >>> config = ToolkitClientConfig(cluster="bluefield", project="my_project", ...)
+            >>> config.create_app_url("/some/app/endpoint")
+            "https://bluefield.cognitedata.com/apps/v1/projects/my_project/some/app/endpoint"
+        """
+        if not endpoint.startswith("/"):
+            endpoint = f"/{endpoint}"
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/client/_toolkit_client.py:112]:
```suggestion
        if not endpoint or not api_version:
            raise RuntimeError("Endpoint and api_version cannot be empty')
        if not endpoint.startswith("/"):
            endpoint = f"/{endpoint}"
```

File: cognite_toolkit/_cdf_tk/storageio/_applications.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MyBitBird) [reply cognite_toolkit/_cdf_tk/storageio/_applications.py:29]: I assume that pagination and more flexible filters would be added to the API, maybe having a TODO here and/or a jira ticket could be nice to make sure we fix it later.

**Review by MyBitBird — COMMENTED** [review]
Looks good, some small comments

File: tests/test_unit/test_cdf_tk/test_storageio/test_applications.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                owner_id="evenOwner" if i % 2 == 0 else "oddOwner",
+                created_time=1700000000000,
+                last_updated_time=1700000000000,
+            )
+            for i in range(20)
+        ]
+    )
+
+
+class TestChartIO:
```
[the remark this answers is not in the document]
Comment (MyBitBird) [reply tests/test_unit/test_cdf_tk/test_storageio/test_applications.py:36]: I'm following up on versions to see if each app has an arbitrary version or they all follow `vX` pattern, but you still need to add tests for when endpoint or/and version is empty/blank

[the remark this answers is not in the document]
Comment (doctrino) [reply tests/test_unit/test_cdf_tk/test_storageio/test_applications.py:36]: I hardcoded version and will add this if I need a flexible version parameter later.

**Review by MyBitBird — DISMISSED** [review]
risk review ok, please also fix typo in the title!

File: cognite_toolkit/_cdf_tk/storageio/_applications.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amorken) [reply cognite_toolkit/_cdf_tk/storageio/_applications.py:29]: Sure there is another way, discard parts of the response if it exceeds what you are willing to handle, and redo the request. Wasteful of course, but blows up in a different way (performance degradation and API/CPU usage) than running out of memory, trashing heaps and/or just dying. I think the Jira issue should be on the charts API itself. Jira is, after all, just a very roided out todo list.
